src/components/ImageView/index.js

```diff
@@ -117,67 +122,41 @@ class ImageView extends PureComponent {
             return;
         }
 
-        this.setState(prevState => ({
-            isZoomed: !prevState.isZoomed,
-            isMouseDown: false,
-        }));
+        // We hold for set isZoomed state if true
+        // Because when set isZoomed state can't getting actual position user clicked
+        this.isZoomed = !this.state.isZoomed;
+        if (this.isZoomed === false) {
```

Comment:
I'm a little confused why this logic is in `onContainerPressOut` and why `this.isZoomed` is needed.
I feel like this can be simplified if it lived in `onContainerPress`, then we wouldn't need this.isZoomed or even need to do multiple setStates.

So wouldn't this  `onContainerPress` code work? (untested)
```js
onContainerPress(e) {
    let scrollX;
    let scrollY;
    if (!this.state.isZoomed && !this.state.isDragging) {
        const {offsetX, offsetY} = e.nativeEvent;

        // We divide the clicked positions by the zoomScale.
        // We need pixel coordinates.
        const delta = this.getScrollOffset(offsetX / this.state.zoomScale, offsetY / this.state.zoomScale);
        scrollX = delta.offsetX;
        scrollY = delta.offsetY;
    }

    if (this.state.isZoomed && this.state.isDragging && this.state.isMouseDown) {
        this.setState({isDragging: false, isMouseDown: false});
    } else {
        // We first zoom and once its done if we are zooming in then we scroll to the location the user clicked.
        this.setState(prevState => ({
            isZoomed: !prevState.isZoomed,
            isMouseDown: false,
        }), () => {
            this.scrollableRef.scrollTop = scrollY;
            this.scrollableRef.scrollLeft = scrollX;
        });
    }
}
```

Reply:
It worked perfectly. I wish I could think of that. Thank you so much. Here is the test video. I will use this code and re-shoot videos for web and desktop.

https://user-images.githubusercontent.com/6093207/168396130-e1290b8c-ac4d-461c-86a1-0891f915043d.mp4

Thanks to this update, we can delete these lines.

https://github.com/Expensify/App/blob/9a8291ba97c381b651b5344d6fa471c51ae70336/src/components/ImageView/index.js#L26

https://github.com/Expensify/App/blob/9a8291ba97c381b651b5344d6fa471c51ae70336/src/components/ImageView/index.js#L115-L124

https://github.com/Expensify/App/blob/9a8291ba97c381b651b5344d6fa471c51ae70336/src/components/ImageView/index.js#L287